src/charm.py

```diff
@@ -117,6 +123,55 @@ def _on_config_changed(self, event: ops.HookEvent) -> None:
         """
         self.change_config(event)
         self._set_workload_version()
+        if self._charm_state.enable_mjolnir:
+            self._enable_mjolnir()
+
+    def _enable_mjolnir(self) -> None:
+        """Enable mjolnir service."""
+        container = self.unit.get_container(SYNAPSE_CONTAINER_NAME)
+        if not container.can_connect():
+            self.unit.status = ops.MaintenanceStatus("Waiting for pebble")
+            return
+        self.model.unit.status = ops.MaintenanceStatus("Configuring Mjolnir")
+        try:
+            admin_username = token_hex(16)
+            admin_user = actions.register_user(container, admin_username, True)
+            admin_access_token = admin_user.access_token
+            mjolnir_user = actions.register_user(
+                container,
+                MJOLNIR_USER,
+                True,
+                str(self._charm_state.server_name),
+                admin_access_token,
+            )
+            # Create (or get) the management room
+            # Add the bot to the management room if we are creating it
+            mjolnir_access_token = mjolnir_user.access_token
+            room_id = synapse.get_room_id(room_name="management", access_token=admin_access_token)
```

Comment:
The room name should probably be a constant

Reply:
Done!